client/src/app/add/add-note.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Router } from '@angular/router';
+import { Note } from '../note';
+import { NotesService } from '../notes.service';
+
+@Component({
+  selector: 'app-add-note',
+  templateUrl: './add-note.component.html',
+  styleUrls: ['./add-note.component.scss']
+})
+export class AddNoteComponent implements OnInit {
+
+  addNoteForm: FormGroup;
+
+  note: Note;
+
+  constructor(private fb: FormBuilder, private noteService: NotesService, private snackBar: MatSnackBar, private router: Router) {
+  }
+
+  // not sure if this name is magical and making it be found or if I'm missing something,
+  // but this is where the red text that shows up (when there is invalid input) comes from
+  add_note_validation_messages = {
```

Comment:
Would you be so kind as to rename this to `addNoteValidationMessages`? The underscores are a little weird.

(You'll need to change add-note.component.html as well.)

Reply:
For the record, the name is not magical. The thing that's making it be found is that you're referencing it from `add-note.component.html`.